src/components/license-activation/data/service.js

```diff
@@ -1,7 +1,25 @@
+import Cookies from 'universal-cookie';
 import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
 import qs from 'query-string';
 
-export function activateLicense(activationKey) {
+export async function activateLicense(activationKey) {
+  // If the user has not refreshed their JWT since they created their account,
+  // we should refresh it so that they'll have appropriate roles (if available),
+  // and thus, have any appropriate permissions when making downstream requests.
+  const loginRefreshUrl = `${process.env.LMS_BASE_URL}/login_refresh`;
+
+  try {
+    await getAuthenticatedHttpClient().post(loginRefreshUrl);
+  } catch (error) {
+    const userIsUnauthenticated = error.response && error.response.status === 401;
+    if (userIsUnauthenticated) {
+      // Clean up the cookie if it exists to eliminate any situation
+      // where the cookie is not expired but the jwt is expired.
+      const cookies = new Cookies();
+      cookies.remove(process.env.ACCESS_TOKEN_COOKIE_NAME);
+    }
+  }
```

Comment:
Is this something worth generalizing? I hope we don't need it anywhere else, but just curious

Reply:
I think ideally we would expose this (or something like it) from frontend-platform/auth: https://github.com/edx/frontend-platform/blob/db97454588dae1af718321b30908279e36bea191/src/auth/AxiosJwtTokenService.js#L54